dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java

```diff
@@ -0,0 +1,294 @@
+import static com.dotmarketing.business.PermissionAPI.PERMISSION_READ;
+import static com.dotmarketing.business.PermissionAPI.PERMISSION_WRITE;
+
+import com.dotcms.contenttype.model.type.BaseContentType;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.enterprise.LicenseUtil;
+import com.dotcms.enterprise.license.LicenseLevel;
+import com.dotcms.publisher.endpoint.bean.PublishingEndPoint;
+import com.dotcms.repackage.com.ibm.icu.text.SimpleDateFormat;
+
+import com.dotmarketing.beans.ContainerStructure;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.factories.MultiTreeAPI;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.portlets.structure.model.Structure;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.tag.model.Tag;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.StringWriter;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.velocity.context.Context;
+
+import com.google.common.collect.Table;
+import com.liferay.portal.model.User;
+
+public class PageContextBuilder {
+    private static PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+
+    final IHTMLPage htmlPage;
+    final User user;
+    final Map<String, Object> ctxMap;
+    final PageMode mode;
+    final Date timeMachine;
+    List<Tag> pageFoundTags;
+
+    final static String WIDGET_PRE_EXECUTE = "WIDGET_PRE_EXECUTE";
+    public PageContextBuilder(IHTMLPage htmlPage, User user, PageMode mode, Date timeMachine)
+            throws DotSecurityException, DotDataException {
+        super();
+        this.htmlPage = htmlPage;
+        this.user = user;
+        this.ctxMap = new HashMap<>();
+        this.mode = mode;
+        this.timeMachine = timeMachine;
+        populateContext();
+        populateContainers();
+    }
+
+
+
+    public PageContextBuilder(IHTMLPage htmlPage, User user, PageMode mode) throws DotSecurityException, DotDataException {
+        this(htmlPage, user, mode, null);
+    }
+
+    private void populateContext() throws DotDataException, DotSecurityException {
+
+
+            ctxMap.put(mode.name(), Boolean.TRUE);
+        
+        
+        
+        
+        
+        
+        // set the page cache var
+        if (htmlPage.getCacheTTL() > 0 && LicenseUtil.getLevel() >= LicenseLevel.COMMUNITY.level) {
+            ctxMap.put("dotPageCacheDate", new java.util.Date());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use the Java 8 Date and Time API instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2143)